src/key_gen/mod.rs

```diff
@@ -727,10 +764,8 @@ impl KeyGen {
         invalid_msg: Vec<u8>,
     ) -> Result<Vec<Message>, Error> {
         if self.phase != Phase::Complaining {
-            return Err(Error::UnexpectedPhase {
-                expected: Phase::Complaining,
-                actual: self.phase,
-            });
+            trace!("To avoid triggering AE pattern, skip this so far");
+            return Ok(Vec::new());
```

Comment:
I'm not sure why we'd want to skip this? (My knowledge of DKG is...low... forgive me!)

Reply:
To make the `Complaint` message handable, we may have to cache All the previous Initial/Proposal/Acknowledgment messages, which will complex the DKG AE pattern a lot.
As, I mentioned before, the current DKG AE is based on a `simplified DKG phases` , which doesn't force the Complaint to be supported.